src/Registries/Statistics.h

```diff
@@ -0,0 +1,833 @@
+		// V 1.9 achievements
+	{"minecraft.sneak_time", "stat.crouchOneCm"},
+	{"minecraft.sleep_in_bed", "stat.sleepInBed"},
+	{"minecraft.aviate_one_cm", "stat.aviateOneCm"}
+};
+
+// V 1.13 stuff
+
+const std::map<AString, int> v1_13StatLookup =
+{
+	{"minecraft.leave_game",                     0},
+	{"minecraft.play_one_minute",                1},
+	{"minecraft.time_since_death",               2},
+	{"minecraft.time_since_rest",                3},
+	{"minecraft.sneak_time",                     4},
+	{"minecraft.walk_one_cm",                    5},
+	{"minecraft.crouch_one_cm",                  6},
+	{"minecraft.sprint_one_cm",                  7},
+	{"minecraft.swim_one_cm",                    8},
+	{"minecraft.fall_one_cm",                    9},
+	{"minecraft.climb_one_cm",                  10},
+	{"minecraft.fly_one_cm",                    11},
+	{"minecraft.walk_under_water_one_cm",       12},
+	{"minecraft.minecart_one_cm",               13},
+	{"minecraft.boat_one_cm",                   14},
+	{"minecraft.pig_one_cm",                    15},
+	{"minecraft.horse_one_cm",                  16},
+	{"minecraft.aviate_one_cm",                 17},
+	{"minecraft.walk_on_water_one_cm",          18},
+	{"minecraft.jump",                          19},
+	{"minecraft.drop",                          20},
+	{"minecraft.damage_dealt",                  21},
+	{"minecraft.damage_taken",                  22},
+	{"minecraft.deaths",                        23},
+	{"minecraft.mob_kills",                     24},
+	{"minecraft.animals_bred",                  25},
+	{"minecraft.player_kills",                  26},
+	{"minecraft.fish_caught",                   27},
+	{"minecraft.talked_to_villager",            28},
+	{"minecraft.traded_with_villager",          29},
+	{"minecraft.eat_cake_slice",                30},
+	{"minecraft.fill_cauldron",                 31},
+	{"minecraft.use_cauldron",                  32},
+	{"minecraft.clean_armor",                   33},
+	{"minecraft.clean_banner",                  34},
+	{"minecraft.interact_with_brewingstand",    35},
+	{"minecraft.interact_with_beacon",          36},
+	{"minecraft.inspect_dropper",               37},
+	{"minecraft.inspect_hopper",                38},
+	{"minecraft.inspect_dispenser",             39},
+	{"minecraft.play_noteblock",                40},
+	{"minecraft.tune_noteblock",                41},
+	{"minecraft.pot_flower",                    42},
+	{"minecraft.trigger_trapped_chest",         43},
+	{"minecraft.open_enderchest",               44},
+	{"minecraft.enchant_item",                  45},
+	{"minecraft.play_record",                   46},
+	{"minecraft.interact_with_furnace",         47},
+	{"minecraft.interact_with_crafting_table",  48},
+	{"minecraft.open_chest",                    49},
+	{"minecraft.sleep_in_bed",                  50},
+	{"minecraft.open_shulker_box",              51}
+};
+
+// V 1.13.1 stuff
+
+const std::map<AString, int> v1_13_1StatLookup =
```

Comment:
This should be a `switch` statement from `enum class Statistic` to Int32/UInt32 (depending on what the protocol uses). Move the protocol-specific mappings into the namespaces in Protocol/Palettes/Palette_1_1X.

Reply:
The compiler will optimise the switch much better than maps

Reply:
good to know - conversion in the way

Reply:
replaced any map possible with switch/case only one is left. but is only used once if the player joins the game and the file is read.

did some research but there seems no way to use something else because switch/case don't support strings in c++ (which makes sense) and if/else has an O(n) in worst case and the map lookup O(log n)

Reply:
The other way is to replace the string descriptions in the json with just numbers which would be faster but hurt the readability quite bad

Reply:
Agreed.